src/java.compiler/share/classes/javax/lang/model/element/Modifier.java

```diff
@@ -51,38 +51,119 @@ public enum Modifier {
 
     // Note java.lang.reflect.Modifier includes INTERFACE, but that's a VMism.
 
-    /** The modifier {@code public} */          PUBLIC,
-    /** The modifier {@code protected} */       PROTECTED,
-    /** The modifier {@code private} */         PRIVATE,
-    /** The modifier {@code abstract} */        ABSTRACT,
+    /**
+     * The modifier {@code public}
+     *
+     * @jls 6.6 Access Control
+     */
+    PUBLIC,
+
+    /**
+     * The modifier {@code protected}
+     *
+     * @jls 6.6 Access Control
+     */
+    PROTECTED,
+
+    /**
+     * The modifier {@code private}
+     *
+     * @jls 6.6 Access Control
+     */
+    PRIVATE,
+
+    /**
+     * The modifier {@code abstract}
+     *
+     * @jls 8.1.1.1 {@code abstract} Classes
+     * @jls 8.4.3.1 {@code abstract} Methods
+     * @jls 9.1.1.1 {@code abstract} Interfaces
+     */
+    ABSTRACT,
+
     /**
      * The modifier {@code default}
+     *
+     * @jls 9.4 Method Declarations
      * @since 1.8
      */
      DEFAULT,
-    /** The modifier {@code static} */          STATIC,
+
+    /**
+     * The modifier {@code static}
+     *
+     * @jls 8.1.1.4 {@code static} Classes
```

Comment:
Do we need to add this?

    @jls 9.1.1.3 {@code static} Interfaces

Reply:
Good catch @pavelrappo ; fixed.